[DPE-6161] Document terminology decisions (#286)
canonical/kafka-operator: README.md

- SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
- Access control management supported with user-provided ACL lists.

> As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of the [Apache ZooKeeper Operator](https://github.com/canonical/zookeeper-operator) for various essential functions.

Reviewer: ```suggestion
As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of [Charmed Apache ZooKeeper](https://github.com/canonical/zookeeper-operator) for various essential functions.
```

### Deployment

> The Apache Kafka and Apache ZooKeeper operators can both be deployed as follows:

Reviewer: ```suggestion
Charmed Apache Kafka and Charmed Apache ZooKeeper can both be deployed as follows:
```

Reviewer: if you really want to use operator, i have noticed that elsewhere you were capitalizing that. Anyhow I would just drop operators (unless we want to only refer to the charm code), but here for instance we refer to both operator code and the workload, so I'd say that we deploy Charmed Apache Kafka and Charmed Apache ZooKeeper

Author: It's not that I specifically wanted to use that term. It was here long before me and out of scope for this PR.  But you're right, it does feel much better with the solution context instead. Good catch!

juju run-action kafka/leader get-admin-credentials --wait
```

> Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, e.g `bin/kafka-config.sh` to update cluster configuration, `bin/kafka-topics.sh` for topic management, and many more! The Apache Kafka Charmed Operator provides these commands for administrators to run their desired cluster configurations securely with SASL authentication, either from within the cluster or as an external client.

Reviewer: ```suggestion
Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, e.g `bin/kafka-config.sh` to update cluster configuration, `bin/kafka-topics.sh` for topic management, and many more! Charmed Apache Kafka provides these commands for administrators to run their desired cluster configurations securely with SASL authentication, either from within the cluster or as an external client.
```

Apache Kafka Charmed Operator sounds weird. Also, it is actually the snap that provides these commands, not the charm code. Anyhow, it is a small difference, I'd just use Charmed Apache Kafka

For example, to list the current topics on the Apache Kafka cluster, run the following command:

```shell
BOOTSTRAP_SERVERS=$(juju run-action kafka/leader get-admin-credentials --wait | grep "bootstrap.servers" | cut -d "=" -f 2)
juju ssh kafka/leader 'charmed-kafka.topics --bootstrap-server $BOOTSTRAP_SERVERS --list --command-config /var/snap/charmed-kafka/common/client.properties'
```

> Note that Charmed Apache Kafka cluster is secure-by-default: when no other application is related to Apache Kafka, listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enabled. If there are no other applications, you can deploy a `data-integrator` charm and relate it to Charmed Apache Kafka to enable listeners.

Reviewer: ```suggestion
Note that Charmed Apache Kafka cluster is secure-by-default: when no other application is related to Charmed Apache Kafka, listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enabled. If there are no other applications, you can deploy a `data-integrator` charm and relate it to Charmed Apache Kafka to enable listeners.
```

> Available Apache Kafka bin commands can be found with:

Reviewer: ```suggestion
Available Charmed Apache Kafka bin commands can be found with:
```

These are available commands in the snap (so the solution), not Apache Kafka (which may even have more or different commands)

## License

> Charmed Apache Kafka is a free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.

Author: ```suggestion
Charmed Apache Kafka is free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.
```
